backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,515 @@
+	return util.NewBadRequestError(errors.New("pipeline default version is unsupported"),
+		"pipeline default version is unsupported when storing in Kubernetes")
+}
+
+func (k *PipelineStoreKubernetes) GetLatestPipelineVersion(pipelineId string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	var latestK8sPipelineVersion *v2beta1.PipelineVersion
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if !k8sPipelineVersion.IsOwnedByPipeline(pipelineId) {
+			continue
+		}
+
+		if latestK8sPipelineVersion == nil || k8sPipelineVersion.CreationTimestamp.Time.After(latestK8sPipelineVersion.CreationTimestamp.Time) {
+			latestK8sPipelineVersion = &k8sPipelineVersion
+
+			continue
+		}
+	}
+
+	if latestK8sPipelineVersion == nil {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", "Latest")
+	}
+
+	return latestK8sPipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersion(pipelineVersionId string) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.getK8sPipelineVersion(context.TODO(), pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionByName(name string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if k8sPipelineVersion.Name == name {
+			pipelineVersion, err1 := k8sPipelineVersion.ToModel()
+			if err1 != nil {
+				return nil, err1
+			}
+
+			return pipelineVersion, nil
+		}
+	}
+	return nil, nil
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionWithStatus(pipelineVersionId string, status model.PipelineVersionStatus) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.GetPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	if pipelineVersion.Status != status {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", pipelineVersionId)
+	}
+
+	return pipelineVersion, nil
+}
+
+func (k *PipelineStoreKubernetes) ListPipelineVersions(pipelineId string, opts *list.Options) (versions []*model.PipelineVersion, totalSize int, nextPageToken string, err error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO())
+	if err != nil {
+		return nil, 0, "", err
+	}
+
+	pipelineVersions := make([]*model.PipelineVersion, 0, len(k8sPipelineVersions.Items))
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if k8sPipelineVersion.IsOwnedByPipeline(pipelineId) {
+			pipelineVersion, err := k8sPipelineVersion.ToModel()
+			if err != nil {
+				return nil, 0, "", err
+			}
+
+			if opts.Filter == nil {
+				pipelineVersions = append(pipelineVersions, pipelineVersion)
+			} else if opts.Filter.FilterK8sPipelineVersions(k8sPipelineVersion) {
```

Comment:
Could you return an error if the unsupported filters are specified? Same for `FilterK8sPipeline`.

I think it's better to let the user know it's unsupported rather than silently ignore their filter.